### ISSUE TITLE
Why was VerifiedSemigroup, VerifiedMonoid, etc removed after 0.9.16?

### ISSUE THREAD

**Issue by Risto-Stevcev** [idris-lang/Idris-dev#3218]
I couldn't find anything in the changelogs. Why was it removed? or did it just get moved to somewhere else?

**Comment by david-christiansen** [comment]
First off, this is really the kind of thing that's more appropriate to the mailing list rather than as a bug report.

We moved them to the `contrib` package, for two reasons:
 1. There weren't interesting instances, and they weren't used in practice. So we didn't think it was appropriate to have them in a place that sends the message that they are the "official" or "best" way to express what they are expressing.
 2. In practice, the `Verified*` interfaces are difficult to use to discharge interesting proof obligations. This is because we don't have a particularly convenient way to talk about equalities of implementation dictionaries. Absent a coherence principle like the one that people advocate for in Haskell, it becomes quite important to track these, because otherwise we have no reason to think that the version for which we proved a law is the same one being used in the code where we want to use that proof. What one really wants to do in Idris is to talk about a *particular implementation* of e.g. `Functor` being verified, not that `Functor List` is verified. The current implementation does not easily support this, and should probably change a bit. Having it in `contrib` makes it more clear that it's not a stable part of the library.
